src/main/host/host.rs

```diff
@@ -533,6 +538,27 @@ impl Host {
         Ref::filter_map(self.processes.borrow(), |processes| processes.get(&id)).ok()
     }
 
+    /// Remove the given process from the Host, if it exists.
+    #[track_caller]
+    pub fn process_remove(&self, id: ProcessId) -> Option<RootedRc<RootedRefCell<Process>>> {
+        self.processes.borrow_mut().remove(&id)
+    }
+
+    /// Borrow the set of processes. Generally this should only be used to
+    /// iterate over the set of processes. e.g. fetching a specific process
+    /// should be done via via `process_borrow`.
+    // TODO: It would be preferable to return an iterator instead of the
+    // collection itself. There has to be an intermediate object though since we
+    // need both the borrowed map of processes, and an iterator that borrows
+    // from that. I suppose we could create an abstract "Iterator factory" and
+    // return that here instead of exposing BTreeMap type.
+    #[track_caller]
+    pub fn processes_borrow(
+        &self,
+    ) -> impl Deref<Target = BTreeMap<ProcessId, RootedRc<RootedRefCell<Process>>>> + '_ {
+        self.processes.borrow()
+    }
```

Comment:
Could you do something like `impl Deref<Target = IntoItererator<Item = (ProcessId, RootedRc<RootedRefCell<Process>>)>` instead?

Reply:
I don't think so, because the iterator needs to *borrow from* the returned object (`Ref<BTreeMap>`), but can't contain both that borrowed object and the internal iterator (without jumping through `unsafe` circular reference hoops). `IntoIterator` won't work here because it consumes `self` instead of borrowing from it.

Reply:
Oh interesting. I can't quite make it work though: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1fa9cdb85ffdbee7b151af1860e86b1b

Reply:
Yeah I don't see how to make it work either. Was worth a try :)